controller/src/main/java/io/pravega/controller/util/RetryHelper.java

```diff
@@ -36,7 +37,8 @@
     public static final Predicate<Throwable> RETRYABLE_PREDICATE = e -> {
         Throwable t = Exceptions.unwrap(e);
         return RetryableException.isRetryable(t) || (t instanceof CheckpointStoreException &&
-                ((CheckpointStoreException) t).getType().equals(CheckpointStoreException.Type.Connectivity));
+                ((CheckpointStoreException) t).getType().equals(CheckpointStoreException.Type.Connectivity)) || 
+                t instanceof IOException;
```

Comment:
when is IOExcpetion thrown?

Reply:
our serializers and deserializers for metadata records can throw IOExceptions. Similarly curator can throw IO exception while reading a znode. Since IO exceptions are generally retriable, so i have included it here.